MoleculeVAE-reImplementation/train.py

```diff
@@ -1,96 +1,330 @@
 import numpy as np
 from rdkit import rdBase, Chem
-print(rdBase.rdkitVersion)
 import pandas as pd
 import os
 import sys
 import hyperparameters
+
 sys.path.append(os.path.join(Chem.RDConfig.RDContribDir, 'SA_Score'))
 import torch.nn as nn
 import torch.utils.data
 import argparse
 import torch.optim as optim
 import torch
-from dataloader import oneHotdecoder
+
 from model import MolecularVAE
+from model import CustomMoleculeDataset
+
+from dataloader import load_dataset
+import torch.utils.data as data
+
+import matplotlib.pyplot as plt
+
+import gif
+import warnings
+
+warnings.filterwarnings("ignore")
+
+import os
+from datetime import datetime
+
+# dd/mm/YY H:M:S
+dt_string = datetime.utcnow().strftime("%d/%m/%Y %H:%M:%S")
+print("date and time =", dt_string)
+
+'''Lists for recording experimental results'''
+# loss_record_list for recording the loss terms [epoch,vae_loss, reconstruction_loss, prediction_loss,process_type]
+record_list = []
+# performance on separate prediction tasks [epoch,'logP_pre', 'qed_pre', 'SAS_pre', 'mae_logP', 'mae_qed', 'mae_SAS']
+separate_prediction_list = []
+# list for record annealing performance [epoch,test_total_loss_with_annealing,test_total_loss_without_annealing,annealing_weights,process_type]
+list_for_record_annealing_performance = []
```

Comment:
Here you are using these global (or at least Module level) variables to record losses over time. I understand the temptation but I think it's better to have them as normal (local variables) and pass them out of your `train` and `test` functions.

Reply:
Thank you for telling me this,I will try some other ways to record them

Reply:
I've tried to summarize them to a class and record the loss value, is that a solution could be used?